src/box/box.cc

```diff
@@ -1835,6 +1841,8 @@ box_promote_qsync(void)
 		diag_set(ClientError, ER_NOT_LEADER, raft->leader);
 		return -1;
 	}
+	if (box_raft_wait_term_broadcasted() != 0)
```

Comment:
Here `raft->term == raft->volatile_term`, because the state is `RAFT_STATE_LEADER` as checked a few lines above. It means the term is already on disk and is already pushed to the relay threads. Doesn't it mean that you can write promote without any waiting and it won't overtake the term update anyway?

Reply:
Thanks for indepth review! The problem is term might be in relay for some time, during that time promote can be written to WAL and broadcasted by relay before term is. AFAIK there is no guarantee that relay will broadcast new term, before it broadcasts promote from wal.

Reply:
Ok, I see, then lets leave it.